### PR TITLE
CI: run FSharpLint on itself

### PR DESCRIPTION
**Pull request by Mersho**

### CODE REVIEW

**Comment by knocte** [comment]
Nits on commit messages:

>FSharpLint.Core: Resolve pascalcase consideration

Use lowercase after ':'.

> build.yml: run Fsharplint on itself

Rather use CI as scope, not build.yml; also please capitalize FSharpLint properly.

> FSharpLint.Core: apply lint warnings

Rather 'respect our own lint rules'.

After you have fixed commit msgs, push, and after that, please rebase your branch.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -30,3 +30,8 @@ jobs:
       run: dotnet restore
     - name: Run build
       run: dotnet fake build -t Test
+    - name: Run Fsharplint on itself
+      run: |
+        cd ./src/FSharpLint.Console
+        dotnet run --framework net5.0 lint ../FSharpLint.Core/FSharpLint.Core.fsproj
+        dotnet run --framework net5.0 lint ./FSharpLint.Console.fsproj
```
Comment (knocte) [inline .github/workflows/build.yml:37]: @Mersho man, look at the scope of the commit msg where you added this change, do you think it's correct?

File: src/FSharpLint.Core/Framework/HintParser.fs [hunk 2]
```diff
@@ -389,10 +389,10 @@ module HintParser =
     let charListToString charList =
         Seq.fold (fun x y -> x + y.ToString()) "" charList
 
-    let pischar chars : Parser<char, 'a> =
+    let pischar chars : Parser<char, 'A> =
```
Comment (bartelink) [inline src/FSharpLint.Core/Framework/HintParser.fs:392]: 'T please! (ot what does A stand for?)

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -30,3 +30,8 @@ jobs:
       run: dotnet restore
     - name: Run build
       run: dotnet fake build -t Test
+    - name: Run Fsharplint on itself
+      run: |
+        cd ./src/FSharpLint.Console
+        dotnet run --framework net5.0 lint ../FSharpLint.Core/FSharpLint.Core.fsproj
```
Comment (knocte) [inline .github/workflows/build.yml:36]: @Mersho what happens if you don't use the --framework flag here? This addition is kinda weird so it needs to be explained in the commit msg

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -30,3 +30,8 @@ jobs:
       run: dotnet restore
     - name: Run build
       run: dotnet fake build -t Test
+    - name: Run Fsharplint on itself
```
Comment (knocte) [inline .github/workflows/build.yml:33]: @Mersho please capitalize our project name properly

File: .github/workflows/build.yml (hunk 3 above)
Comment (knocte) [reply .github/workflows/build.yml:36]: interesting!

**Comment by Mersho** [comment]
Closed in favor of https://github.com/fsprojects/FSharpLint/pull/628

**Pull request closed without merging**